api/v1alpha1/adminnetworkpolicy_types.go

```diff
@@ -0,0 +1,309 @@
+	Conditions []metav1.Condition `json:"status"`
+}
+
+// AdminNetworkPolicySubject defines what objects the policy selects.
+// Exactly one of the `NamespaceSelector` or `NamespaceAndPodSelector` pointers
+// should be set.
+type AdminNetworkPolicySubject struct {
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector"`
+	// +optional
+	NamespaceAndPodSelector *NamespacedPodSubject `json:"namespaceAndPodSelector"`
+}
+
+// NamespacedPodSubject allows the user to select a given set of pod(s) in
+// selected namespace(s)
+type NamespacedPodSubject struct {
+	// This field follows standard label selector semantics; if present but empty,
+	// it selects all Namespaces.
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector"`
+
+	// Used to explicitly select pods within a namespace; if present but empty,
+	// it selects all Pods.
+	PodSelector *metav1.LabelSelector `json:"podSelector"`
+}
+
+// AdminNetworkPolicySpec defines the desired state of AdminNetworkPolicy
+type AdminNetworkPolicySpec struct {
+	// Priority is an int32 value bound to 0 - 1000, the lowest positive priority,
+	// "1" corresponds to the highest importance, while higher priorities have
+	// lower importance. An ANP with a priority of "0" will be evaluated after all
+	// positive priority AdminNetworkPolicies and standard NetworkPolicies.
+	// This field is NOT optional.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=1000
+	Priority *int32 `json:"priority"`
+
+	// Subject defines the pods to which this AdminNetworkPolicy applies.
+	// This field is NOT optional.
+	Subject AdminNetworkPolicySubject `json:"subject"`
+
+	// List of Ingress rules to be applied to the selected objects.
+	// A total of 100 rules will be allowed per each network policy instance,
+	// this rule count will be calculated as the total summation of the
+	// Ingress and Egress rules in a single AdminNetworkPolicy Instance.
+	// If this field is empty then this AdminNetworkPolicy has no effect on
+	// ingress traffic.
+	// +optional
+	Ingress []AdminNetworkPolicyIngressRule `json:"ingress,omitempty"`
+
+	// List of Egress rules to be applied to the selected objects.
+	// A total of 100 rules will be allowed per each network policy instance,
+	// this rule count will be calculated as the total summation of the
+	// Ingress and Egress rules in a single AdminNetworkPolicy Instance.
+	// If this field is empty then this AdminNetworkPolicy has no effect on
+	// egress traffic.
+	// +optional
+	Egress []AdminNetworkPolicyEgressRule `json:"egress,omitempty"`
+}
+
+// AdminNetworkPolicyIngressRule describes an action to take on a particular
+// set of traffic destined for pods selected by an AdminNetworkPolicy's
+// Subject field. The traffic must match both ports and from.
+type AdminNetworkPolicyIngressRule struct {
+	// Name is an identifier for this rule, that should be no more than 100 characters
+	// in length.
```

Comment:
Agreed....However, I thinl make it much easier for any user oriented tooling we build around ANP to report issues back to the user, and even further future looking in terms of ANP logging, I think this could help considerably.  

Immediately I think folks would use this to help Admins organize ANPs for better readability/ organization 

Although there isn't a true "we NEED this because..." rational for this, I think it would be nice to have and I don't really see a large downside to having it.

Reply:
clarification, "The Why" needs to be explained in the documentation.